File: atsa/src/main/java/jp/kaiz/atsassistmod/item/TrainProtectionSelectorItem.java
package jp.kaiz.atsassistmod.item;

import net.minecraft.world.InteractionHand;
import net.minecraft.world.InteractionResult;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;

/**
 * Opens the train-protection selector screen on use. The original opened a GUI via
 * {@code player.openGui}; the screen itself is wired in the GUI stage (TODO).
 */
public class TrainProtectionSelectorItem extends Item {
    public TrainProtectionSelectorItem(Properties properties) {
        super(properties);
    }

    @Override
    public InteractionResult use(Level level, Player player, InteractionHand hand) {
        ItemStack stack = player.getItemInHand(hand);
        if (level.isClientSide()) {
            jp.kaiz.atsassistmod.client.ATSAModClientHooks.openTrainProtectionSelector();
        }
        return InteractionResult.SUCCESS;
    }
}

File: atsa/src/main/kotlin/jp/kaiz/atsassistmod/item/TrainProtectionSelectorItem.kt
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright © 2026 mirukuneko and RealTrainModRenewed contributors
package jp.kaiz.atsassistmod.item

import jp.kaiz.atsassistmod.client.ATSAModClientHooks
import net.minecraft.world.InteractionHand
import net.minecraft.world.InteractionResult
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.Item
import net.minecraft.world.level.Level

/**
 * Opens the train-protection selector screen on use.
 */
open class TrainProtectionSelectorItem(properties: Properties) : Item(properties) {
    override fun use(level: Level, player: Player, hand: InteractionHand): InteractionResult {
        if (level.isClientSide) {
            ATSAModClientHooks.openTrainProtectionSelector()
        }
        return InteractionResult.SUCCESS
    }
}
